ethereum/contracts/OutChannel.sol

```diff
@@ -0,0 +1,14 @@
+ // SPDX-License-Identifier: MIT
+pragma solidity >=0.6.2;
+
+// OutChannel contains methods that all outgoing channels must implement
+abstract contract OutChannel {
+
+    event NewMessage(uint256 nonce, address senderAddress, string targetApplicationId, bytes payload);
```

Comment:
`nonce` in here implies that we use nonces as replay protection on every `OutChannel`. Can we fing a more generic term?  Maybe `replayProtection`?

Reply:
nonce is the generic term :P - https://en.wikipedia.org/wiki/Cryptographic_nonce